### ISSUE TITLE
Prompt for a 2FA code before disabling 2FA

### ISSUE THREAD

**Issue by rfk** [mozilla/fxa#6055]
This came in via a mailing list, I couldn't find a definitive bug (not even a RESOLVED/FIXED one) on this topic so I figured I'd file it as a new issue.

From the submitter:

"""
After opening my Firefox account's settings ( https://accounts.firefox.com/settings ) I was able to disable two factor authentication
without providing my 2fa code first. This poses a security risk and is in direct violation of the OWASP Application Security Verification Standard 4.0 [p. 31 f., 1].

Please consider changing this behavior.
"""

In other words, this user expected that they would need to enter a 2FA code to confirm they are in control of the 2FA device, before they would be allowed to disable 2FA on the account.

I wasn't able to find the specific advice mentioned in the OWASP standard (the things on page 31 don't *seem* related from my initial read) but I'll admit I didn't try super hard.

**Comment by phenomax** [comment]
Original quotation:

`[1] https://owasp.org/www-pdf-archive/OWASP_Application_Security_Verification_Standard_4.0-en.pdf`

Pages 31 + 32 describe a possible attack, changing sensitive account settings (e.g. disable two factor auth) after obtaining a valid login session without verifying a two factor token.
Hence `3.7.1` requires
```
Verify the application ensures a valid login session or requires re-authentication or secondary verification before allowing any sensitive transactions or account modifications.
```
